### ISSUE TITLE
Filesystem errors during module compilation

### ISSUE THREAD

**Issue by mikolaj6r** [denoland/deno#6363]
Console output:
```bash
mikuus4@DELL-Inspiron-5548:~/projects/denjucks$ deno run test.ts
error: No such file or directory (os error 2)
mikuus4@DELL-Inspiron-5548:~/projects/denjucks$ deno run mod.js
Compile file:///home/mikuus4/projects/denjucks/mod.js
mikuus4@DELL-Inspiron-5548:~/projects/denjucks$ deno run test.ts
undefined
```

Looks like sometimes during compile step, Deno can't access imported files. Especially these with longer tree like: './deps/path/std/path/mod.ts'. 

The best way to reproduce this error is to clone this repo: [https://github.com/mikolaj6r/deno-error-poc](https://github.com/mikolaj6r/deno-error-poc) and follow steps in readme.

**Comment by mikolaj6r** [comment]
After updating to v1.1.1:
```bash
Compile file:///app/mod.ts
error: Failed to get compiled source code of https://deno.land/x/denjucks/src/deps/path/std/path/mod.ts.
Reason: No such file or directory (os error 2)
```

**Comment by bartlomieju** [comment]
Thanks for reporting the issue, could you boil down reproduction to some smaller example? 

Ref #6057

**Comment by joeGermany** [comment]
@bartlomieju 
I don't know if this is what you are looking for, but a smaller example could be:
```
import { Application, send, Context, RouterContext, Router } from "./deps.ts";
import { staticFileMiddleware } from './staticFileMiddleware.ts';

const app = new Application();
const router = new Router();

app.use(router.routes());
app.use(router.allowedMethods());
app.use(staticFileMiddleware);

router.get('/(.*)', async (context: RouterContext) => {
    console.log(context.params[0]);
    await send(context, context.request.url.pathname, {
        root: `${Deno.cwd()}/public`,
        index: "index.html",
    });
});

app.addEventListener('listen', ({hostname, port, secure}) => {
    console.log(`Listening on ${secure ? 'https://' : 'http://'}${hostname || 'localhost'}:${port}`)
});

app.addEventListener('error', e => {
    console.log(e.error);
});

await app.listen({port: 8000});
```

If you try to go to `localhost:8000` everything plays out according to my desires. But when I try to type in the URL `localhost:8000/login`, **I get the following error:**

NotFoundError: The system cannot find the file specified. (os error 2)
    at createHttpError (httpError.ts:129:10)
    at send (send.ts:147:13)
    at async serverOak.ts:15:5
    at async dispatch (middleware.ts:39:7)
    at async dispatch (middleware.ts:39:7)
    at async dispatch (middleware.ts:39:7)
    at async Application.#handleRequest (application.ts:250:9)

**Comment by bartlomieju** [comment]
@joeGermany thanks for reporting, but issue you're experiencing is completely different - it comes from your runtime code and not Deno's builtin compiler. I'm pretty sure `await send(context, context.request.url.pathname` might be the problematic line. I suggest printing out contents of the pathname and making sure it's a valid file path.

**Comment by joeGermany** [comment]
@bartlomieju Sorry for reporting misleading clues. However, could you help me with a question I have (although it is not directly related to the issue described here)?
Could you kindly clarify what the 2nd argument requires in the send function? I mean what am I supposed to put inside of it (can I put something other than `context.request.url.pathname`).
What I am trying to accomplish in my code is to always redirect the user to index.html (without displaying it in the URL in the browser). Do you have any idea how this could be accomplished?

**Comment by bartlomieju** [comment]
I think this issue was fixed as part of #6057
